internal/types/base.go

```diff
@@ -0,0 +1,13 @@
+package types
+
+import (
+	"bytes"
+	"os"
+)
+
+// KeepalivedCollector is an interface for implementing multiple collector modes
+type KeepalivedCollector interface {
```

Comment:
This should be inside of the `collector` package.

Reply:
Can you please suggest a new name for it because it will conflict with https://github.com/cafebazaar/keepalived-exporter/pull/55/files#diff-931128854bf930aa328bc712a7074b59R17

Reply:
I think ‍`Collector` is good enough for now.